02_knight_and_princess/xieyu_wang/main.cpp

```diff
@@ -106,13 +106,14 @@ class Graph {
   }
 
  private:
-  Vertex& getVertex(int id) const {
-    for (const auto& vertex : vertices_) {
+  Vertex& getVertex(int id) {
+    assert(hasVertex(id) && "Vertex doesn't exist");
+    for (auto& vertex : vertices_) {
       if (vertex.id == id) {
         return (Vertex&)vertex;
       }
     }
-    return const_cast<Vertex&>(vertices_.at(-1));
+    exit(-1);
```

Comment:
It is better to throw exceptions with description, something like this:
```suggestion
  throw std::runtime_error("Unreachable code");
```

This is actually a good showcase why `std::vector` is not the best structure to use here. `std::unordered_map` will be a much better to avoid search for an element.